### ISSUE TITLE
mmlu benchmark scores

### ISSUE THREAD

**Issue by SeunghyunSEO** [facebookresearch/lingua#5]
hi guys, thanks for sharing high quality hackable codebase !
ive just wondered how 7B llama with dclm 1T tokens can achieve 60++ mmlu score.
to my best knowledge, it should consume enough flops (like 10^22++) for emergent ability (mmlu like tasks), and apple dclm 7b model achieve 63 fewshot mmlu score at 2.5T tokens. 
is there any special source for your training? 
i saw some details like model init method (you guys use mitchell init  from olmo and torchtitan),
but none of ml methods like parametrization can beat performance vs flops trend so hard, i guess.
or is there any difference in evaluation logic? (system propmt or something)

**Comment by BadrYoubiIdrissi** [comment]
Hi Seunghyun ! Thank you for your interest in lingua :) I think we're approx reproducing DCLM's results ! At 200B tokens we're at 48.64 wherease at 280B tokens they are at 48.9. When you do an exponential decay fit on their results (which is a bad model but just for the sake of the argument) you get something above 60 also at 1T tokens. So i think it's just that the DCLM data is really good for MMLU !
![Exponential Decay Fit (Trillions of Tokens vs MMLU)](https://github.com/user-attachments/assets/00db4e8b-aa9e-4ebc-a344-8e2b06b86617)

**Comment by SeunghyunSEO** [comment]
thank you so much @BadrYoubiIdrissi , my reproduction on dclm has different batch size, so optimization steps differs in same tokens i guess. i will reproduce baseline with lingua thank you!

**Comment by BadrYoubiIdrissi** [comment]
Please let me know how it goes! Full disclosure our results were computed with an internal evaluation library so there might still be a difference. Ill try to rerun evals with lm-eval harness and see if there are any differences.

**Comment by SeunghyunSEO** [comment]
@BadrYoubiIdrissi im sry to bother you, may i ask details for 200B training?
i want to know whether you guys train actually 1T but eval 200B point then you got 48 mmlu or training horizon was just 200B.
would be very much appreciate if you let me know peak learning rate, adam betas, weight decay, batch size, warmup steps

**Comment by mathuvu** [comment]
The 200B result is a training on his own, not a mid evaluation point during training of the 1T. The exact config file to reproduce this run can be found here: [llama_7B.yaml](https://github.com/facebookresearch/lingua/tree/main/apps/main/configs/llama_7B.yaml)

This run has been done on 256 GPUs but in principle you can run it on less GPUs by increasing the `grad_acc_steps` parameter (for examples the exact same config on 128 GPUs but `grad_acc_steps: 2`).

**Comment by SeunghyunSEO** [comment]
![dclm_reproduce](https://github.com/user-attachments/assets/bb3a073b-0a1b-43cc-a437-33ec82990149)

@mathuvu @BadrYoubiIdrissi 
hi guys, here's my reproduction result but unfortunately it's not trained with lingua (i used my own codebase) and there is little differences.
i used llama3 8B setup (noam arch GQA/rope/swiglu), so it's ffn size is slightly bigger than your 7B MHA setup and i use standard fan-in initialization with global depth init (but lingua [has a little bug](https://github.com/facebookresearch/lingua/issues/39)).
but i match peak lr, batch size with yours.

**Comment by craffel** [comment]
> The 200B result is a training on his own, not a mid evaluation point during training of the 1T. The exact config file to reproduce this run can be found here: [llama_7B.yaml](https://github.com/facebookresearch/lingua/blob/a38fe06de66fd09c56135dc77bd5fa36ff4febc9/apps/main/configs/llama_7B.yaml)
> 
> This run has been done on 256 GPUs but in principle you can run it on less GPUs by increasing the `grad_acc_steps` parameter (for examples the exact same config on 128 GPUs but `grad_acc_steps: 2`).

Hi @mathuvu ! I am trying to match your baseline results on 64 H100s. When I change grad_acc_steps to 4 (which is what your comment implies is correct), I get an OOM right away. I'm assuming this isn't expected? Anything I can do to debug or any tips?

**Comment by Hannibal046** [comment]
> > The 200B result is a training on his own, not a mid evaluation point during training of the 1T. The exact config file to reproduce this run can be found here: [llama_7B.yaml](https://github.com/facebookresearch/lingua/blob/a38fe06de66fd09c56135dc77bd5fa36ff4febc9/apps/main/configs/llama_7B.yaml)
> > This run has been done on 256 GPUs but in principle you can run it on less GPUs by increasing the `grad_acc_steps` parameter (for examples the exact same config on 128 GPUs but `grad_acc_steps: 2`).
> 
> Hi [@mathuvu](https://github.com/mathuvu) ! I am trying to match your baseline results on 64 H100s. When I change grad_acc_steps to 4 (which is what your comment implies is correct), I get an OOM right away. I'm assuming this isn't expected? Anything I can do to debug or any tips?

Hi, can you try to comment out this code and see if it still gets OOM?
https://github.com/facebookresearch/lingua/blob/d61a3f0139b00d74604828b7303c590cf841cfd5/apps/main/train.py#L417-L418

---
A relevant issue: https://github.com/pytorch/torchtitan/issues/292

**Comment by craffel** [comment]
Thanks for the reply! I had flagged that line as possibly problematic but didn't fully understand what it was doing, so didn't try removing it. The issue is helpful for context. I will give it a try and update with whether it avoids OOMs and whether it decreases throughput.

**Comment by craffel** [comment]
That did the trick! Interestingly throughput is actually higher now. So a win-win. Thanks.
